lib/irgen.ml

```diff
@@ -44,7 +44,17 @@ let translate (binds, sfuncs): L.llmodule =
     in List.fold_left add_func StringMap.empty sfuncs
   in
 
+  let function_decls : (L.llvalue * sfunc) StringMap.t =
+    let func_decl m func =
+      let name = func.sfname
+      and param_types = 
+        Array.of_list (List.map (fun (t,_) -> lltype_of_dtype t) func.sparams)
+            in let ftype = L.function_type (lltype_of_dtype func.srtype) param_types in
+            StringMap.add name (L.define_function name ftype mdl, func) m in
+          List.fold_left func_decl StringMap.empty sfuncs in
```

Comment:
I think this is the same as `all_funcs` right above

Reply:
oh oopsie

Reply:
removed lol